Fix typos and add footnotes

kfrn opened the pull request:
Nice guide! I made a couple of small changes, if you are interested.

yangshun commented on README.md:
```diff
@@ -53,15 +53,15 @@ The benefits:
 While traditional server-side rendered apps are still a viable option, a clear client-server separation scales better for larger engineering teams, as the client and server code can be developed and released independently. This is especially so at Grab when we have multiple client apps hitting the same API server.
 
-As web developers are now building apps rather than pages, organization of client-side JavaScript has become increasingly important. In server-side rendered pages, it is common to use snippets of jQuery to add user interactivity to each page. However, when building large apps, just jQuery insufficient. After all, jQuery is primarily a library for DOM manipulation and it's not a framework; it does ...
+As web developers are now building apps rather than pages, organization of client-side JavaScript has become increasingly important. In server-side rendered pages, it is common to use snippets of jQuery to add user interactivity to each page. However, when building large apps, just is jQuery insufficient. After all, jQuery is primarily a library for DOM manipulation and it's not a framework; it d ...
```
I think this part should be `just jQuery is insufficient` instead

kfrn replied:
Yes, sorry!

yangshun commented on README.md:
```diff
@@ -53,15 +53,15 @@ The benefits:
+As web developers are now building apps rather than pages, organization of client-side JavaScript has become increasingly important. In server-side rendered pages, it is common to use snippets of jQuery to add user interactivity to each page. However, when building large apps, just is jQuery insufficient. After all, jQuery is primarily a library for DOM manipulation and it's not a framework; it d ...
 
-JavaScript frameworks have been created to provide higher-level abstractions over the DOM, allowing you to keep state in memory, out of the DOM. Using frameworks also brings the benefits of reusing recommended concepts and best practices for building apps. A new engineer on the team who is unfamiliar with the code base, but has experience with a framework, will find it easier to understand the co ...
+JavaScript frameworks have been created to provide higher-level abstractions over the DOM, allowing you to keep state in memory, out of the DOM. Using frameworks also brings the benefits of reusing recommended concepts and best practices for building apps. A new engineer on the team who is unfamiliar with the code base, but has experience with a framework, will find it easier to understand the co ...
```
I don't meant for the use of "he" to mean the male gender... How about changing it to "they"?

kfrn replied:
👍

yangshun commented on README.md:
```diff
@@ -53,15 +53,15 @@ The benefits:
 
 The downsides:
 
-- Heavier initial page load due to loading of framework, app code, and assets required for multiple pages [^1].
```
I left it like this because it's the Jekyll footnote format. But I guess I should make it work here too. Could you help change it to use `<sup>[1](#ft-1)</sup>` instead? Using `*` will be kind of unscalable if we have more footnotes in future.

yangshun commented on README.md:
```diff
@@ -388,5 +388,6 @@ As the front end ecosystem grows, we are actively exploring, experimenting and e
 
-[^1]: This can be solved via [webpack code splitting](https://webpack.js.org/guides/code-splitting/).
-[^2]: [Universal JS](https://medium.com/@mjackson/universal-javascript-4761051b7ae9) to the rescue!
+<a href="#ft-1" name="ft-1">\*</a> This can be solved via [webpack code splitting](https://webpack.js.org/guides/code-splitting/).
```
I believe the markdown anchor syntax would work too. For reference, this is how it's supposed to look like: http://engineering.grab.com/grabs-front-end-study-guide

kfrn commented:
Changes made! I can't believe I made a typo while trying to fix a typo (_"just is jQuery insufficient"_ - !!!).  
I think it's nicer not to address only men. Changed to the gender-neutral they, as you suggested.    
I changed the footnotes, in the end just using HTML footnotes. Feel free to change to a markdown style if you prefer :)

yangshun commented:
Awesome, thank you! Shipped! (:
